diff --git a/cfitsio.pc.in b/cfitsio.pc.in
index 7cc25c6..3ab109f 100644
--- a/cfitsio.pc.in
+++ b/cfitsio.pc.in
@@ -7,6 +7,6 @@ Name: cfitsio
 Description: FITS File Subroutine Library
 URL: https://heasarc.gsfc.nasa.gov/fitsio/
 Version: @CFITSIO_MAJOR@.@CFITSIO_MINOR@.@CFITSIO_MICRO@
-Libs: -L${libdir} -lcfitsio
+Libs: -L${libdir} -lcfitsio -lcurl
 Libs.private: -lm @LIBS@
 Cflags: -I${includedir}
