diff --git a/utilities/speed.c b/utilities/speed.c
index 210994e..3086dae 100644
--- a/utilities/speed.c
+++ b/utilities/speed.c
@@ -2,7 +2,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <time.h>
+#if defined(unix) || defined(__unix__)  || defined(__unix)
 #include <sys/time.h>
+#endif
 
 /*
   Every program which uses the CFITSIO interface must include the
@@ -507,6 +509,7 @@ void printerror( int status)
 /*--------------------------------------------------------------------------*/
 int marktime( int *status)
 {
+#if defined(unix) || defined(__unix__)  || defined(__unix)
     double telapse;
     time_t temp;
     struct  timeval tv;
@@ -535,11 +538,17 @@ int marktime( int *status)
 	startsec = tv.tv_sec;
         startmilli = tv.tv_usec/1000;
 
+#else
+    startsec = 0;
+    startmilli = 0;
+    scpu = clock();
+#endif
     return( *status );
 }
 /*--------------------------------------------------------------------------*/
 int gettime(double *elapse, float *elapscpu, int *status)
 {
+#if defined(unix) || defined(__unix__)  || defined(__unix)
         struct  timeval tv;
 	int stopmilli;
 	long stopsec;
@@ -558,5 +567,11 @@ int gettime(double *elapse, float *elapscpu, int *status)
 /*    *elapse = difftime(finish, start) + 0.5; */
     *elapscpu = (ecpu - scpu) * 1.0 / CLOCKTICKS;
 
+#else
+    /* set the elapsed time the same as the CPU time on Windows machines */
+    ecpu = clock();
+    *elapscpu = (float) ((ecpu - scpu) * 1.0 / CLOCKTICKS);
+    *elapse = *elapscpu;
+#endif
     return( *status );
 }
